h2o-core/src/main/java/water/fvec/Vec.java

```diff
@@ -371,12 +371,20 @@ public static Vec makeCon(double x, long len, int log_rows_per_chunk) {
     return makeCon(x,len,log_rows_per_chunk,true);
   }
 
+  private static int VALUE_MAX = 1<<28; //original = 1<<28 = Value.MAX / 8
   /**
    * Make a new constant vector with minimal number of chunks. Used for importing SQL tables.
    *  @return New constant vector with the given row count. */
   public static Vec makeCon(long totSize, long len) {
-    int safetyInflationFactor = 8;
-    int nchunks = (int) Math.max(safetyInflationFactor * totSize / Value.MAX , 1);
```

Comment:
Why do you need to change this? I don't think this is a step in the right direction because you are losing the relationship with Value.MAX. If Value.MAX is changed in the future it won't be reflected here resulting in a bug.

Reply:
I'll restore the default to Value.MAX, I was much more intrigued by the so-called safety factor that mainly looks like a magic number without proper rationale outside artificially changing the Value.MAX

Reply:
I think it is a little confusing to have 2 makeCon() methods that only differ in type long vs int.

This will break existing code that is calling the old method

     int x = 17;
     Vec.makeCon(42.0, x);

Will start behaving different!

Reply:
fair enough as the int and long versions have a different semantic besides: will rename the new one to sth like makeConN to hint that the size N of the vector is fixed and defined by param nchunks

Reply:
closed